spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string createStatement =
+                $"CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) "
+                + "STORING (MarketingBudget)";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.UpdateDatabaseDdlAsync(
+                    new DatabaseName(projectId, instanceId, databaseId),
+                new[] { createStatement });
+            // Poll until the returned long-running operation is complete.
+            var updateDatabaseResponse =
+                await createDatabaseRequest.PollUntilCompletedAsync();
+            return updateDatabaseResponse.Result;
+            // [END create_storing_index]
+        }
+
+        public static async Task<object> AddColumnAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START add_column]
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string alterStatement =
+                $"ALTER TABLE Albums ADD COLUMN MarketingBudget INT64";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.UpdateDatabaseDdlAsync(
+                    new DatabaseName(projectId, instanceId, databaseId),
+                new[] { alterStatement });
+            // Poll until the returned long-running operation is complete.
+            var updateDatabaseResponse =
+                await createDatabaseRequest.PollUntilCompletedAsync();
+            Console.WriteLine("Added the MarketingBudget column.");
+            return updateDatabaseResponse.Result;
+            // [END add_column]
+        }
+
+        public static async Task QuerySampleDataAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/"
+            + $"{instanceId}/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "

```

Comment:
we've put a lot of work into reader.GetFieldValue<T>("columnname").
where T is the type you want the column converted to.  This is a better approach than using the default type we choose for you (via the bracket operator) because if you ever do a schema change, your code has a good chance of still working.
eg.
 you store a column as FLOAT64, but only ever store ints.
 you get the value via:  reader.GetFieldValue<long>("widget_count");
 realizing your error, you switch the column type in spanner to "INT64" -- your code will work without modification.
 However, if you had done long count = Convert.ToInt64((double)reader["widget_count"]);  it will result in a cast exception because the default type we return will change to long.